src/cascadia/TerminalControl/TSFInputControl.cpp

```diff
@@ -388,24 +364,18 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     // - <none>
     void TSFInputControl::_textUpdatingHandler(CoreTextEditContext sender, const CoreTextTextUpdatingEventArgs& args)
     {
-        const auto incomingText = args.Text();
-        const auto range = args.Range();
-
         try
         {
-            // When a user deletes the last character in their current composition, some machines
-            // will fire a CompositionCompleted before firing a TextUpdating event that deletes the last character.
-            // The TextUpdating will have a lower StartCaretPosition, so in this scenario, _activeTextStart
-            // needs to update to be the StartCaretPosition.
-            // A known issue related to this behavior is that the last character that's deleted from a composition
-            // will get sent to the terminal before we receive the TextUpdate to delete the character.
-            // See GH #5054.
-            _activeTextStart = ::base::ClampMin(_activeTextStart, ::base::ClampedNumeric<size_t>(range.StartCaretPosition));
+            const auto incomingText = args.Text();
+            const auto range = args.Range();
 
             _inputBuffer = _inputBuffer.replace(
                 range.StartCaretPosition,
-                ::base::ClampSub<size_t>(range.EndCaretPosition, range.StartCaretPosition),
+                range.EndCaretPosition - range.StartCaretPosition,
                 incomingText);
+            _selection = args.NewSelection();
+            // GH#5054: Pressing backspace might move the caret before the _activeTextStart.
+            _activeTextStart = ::base::ClampMin(_activeTextStart, ::base::ClampedNumeric<size_t>(range.StartCaretPosition));

```

Comment:
I don't think "some machines will fire a CompositionCompleted before firing a TextUpdating event". As far as I can see, all machines do that, but it depends on the circumstances.
I've thus changed the comment to just be about backspacing.